src/neuroconv/datainterfaces/image/imageinterface.py

```diff
@@ -0,0 +1,269 @@
+                    items=dict(type="string"),
+                    description="List of paths to image files to be converted",
+                ),
+                folder_path=dict(
+                    type="string",
+                    description="Path to folder containing images to be converted. Used if file_paths not provided.",
+                ),
+            ),
+        )
+
+    def __init__(
+        self,
+        file_paths: Optional[List[Union[str, Path]]] = None,
+        folder_path: Optional[Union[str, Path]] = None,
+        images_location: Literal["acquisition", "stimulus"] = "acquisition",
+        verbose: bool = True,
+    ):
+        """
+        Initialize the ImageInterface.
+
+        Parameters
+        ----------
+        file_paths : list of Union[str, Path], optional
+            List of paths to image files to be converted
+        folder_path : Union[str, Path], optional
+            Path to folder containing images to be converted. Used if file_paths not provided.
+        images_location : Literal["acquisition", "stimulus"], default: "acquisition"
+            Location to store images in the NWB file
+        verbose : bool, default: True
+            Whether to print status messages
+        """
+        if file_paths is None and folder_path is None:
+            raise ValueError("Either file_paths or folder_path must be provided")
+
+        if file_paths is not None and folder_path is not None:
+            raise ValueError("Only one of file_paths or folder_path should be provided")
+
+        self.file_paths = file_paths
+        self.folder_path = folder_path
+        self.images_location = images_location
+
+        super().__init__(
+            verbose=verbose, file_paths=file_paths, folder_path=folder_path, images_location=images_location
+        )
+
+        # Process paths
+        if folder_path is not None:
+            folder = Path(folder_path)
+            if not folder.exists():
+                raise ValueError(f"Folder {folder} does not exist")
+
+            # Get all image files in folder
+            file_paths = []
+            for suffix in self.associated_suffixes:
+                file_paths.extend(folder.glob(f"*{suffix}"))
+
+            if not file_paths:
+                raise ValueError(f"No image files found in {folder}")
+
+            self.file_paths = [str(Path(p).absolute()) for p in file_paths]
+        else:
+            self.file_paths = [str(Path(p).absolute()) for p in file_paths]
+
+            # Validate paths
+            for path in self.file_paths:
+                if not Path(path).exists():
+                    raise ValueError(f"File {path} does not exist")
+
+    def get_metadata(self) -> DeepDict:
+        """Get metadata for the images."""
+        metadata = super().get_metadata()
+
+        # Add basic metadata about the images
+        metadata["Images"] = dict(description="Images loaded through ImageInterface", num_images=len(self.file_paths))
```

Comment:
why would you need to indicate the number of images?

Reply:
We don't as far as I see. I uncritically accepted this from cline I think.

Reply:
Changed, I also added the correct headers on the `rst` that you mentioned.